plugins/outputs/quix/quix.go

```diff
@@ -109,13 +109,18 @@ func (q *Quix) Connect() error {
 			return fmt.Errorf("unsupported SASL mechanism: %s", quixConfig.SaslMechanism)
 		}
 
-		// Certificate
-		certPool := x509.NewCertPool()
-		if !certPool.AppendCertsFromPEM(quixConfig.cert) {
-			return errors.New("appending CA cert to pool failed")
+		// Certificate (optional)
+		if len(quixConfig.cert) > 0 {
+			certPool := x509.NewCertPool()
+			if !certPool.AppendCertsFromPEM(quixConfig.cert) {
+				return errors.New("appending CA cert to pool failed")
+			}
+			cfg.Net.TLS.Enable = true
```

Comment:
TLS should be enabled unconditionally, shouldn't it?

Reply:
You are right.